crypto/schnorr-adaptor-signature/types.go

```diff
@@ -0,0 +1,173 @@
+package schnorr_adaptor_signature
+
+import (
+	"bytes"
+	"encoding/hex"
+	"fmt"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+)
+
+const (
+	ModNScalarSize       = 32
+	FieldValSize         = 32
+	JacobianPointSize    = 33
+	AdaptorSignatureSize = JacobianPointSize + ModNScalarSize
+)
+
+type AdaptorSignature struct {
+	R0 btcec.JacobianPoint
```

Comment:
I wonder wheter it would better to represent this value as `btcec.PublicKey` which already have a lot of helpers like:
- `SerializeCompressed`
- or `btcec.ParsePubKey`

Reply:
In theory yes, but this might conflate between keys and underlying points representation, and deviate a bit from the spec. So I recommend we use point here and keep it aligned with spec. Wdyt?

Reply:
hmm makes sense 👍